feat: scaffold new process applications documentation (#5416)
camunda/camunda-docs: docs/components/modeler/desktop-modeler/process-applications/process-application.md

---
id: process-application
title: Process applications in Desktop Modeler
description: In Desktop Modeler a process application is a folder that contains a .process-application file and a set of related files you can work on and deploy as a single bundle.
---

import OverlayImg from './img/overlay.png'

> In Desktop Modeler a [process application](../../../concepts/process-applications.md) is a folder that contains a `.process-application` file and a set of related files you can work on and deploy as a single bundle. These files define the automation logic but are often part of a larger project (e.g., a Java application).

Reviewer: ```suggestion
In Desktop Modeler a [process application](../../../concepts/process-applications.md) is a folder that contains a `.process-application` file and a set of Camunda 8 assets that you work on and deploy as a single bundle. Typically the process application also contains custom implementation logic, e.g. your job workers, implemented in a Java application.
```

Author: I find our use of so many different terminologies confusing. We have files, resources, assets. I'll leave it up to the technical writer to decide.

> For instance, a consumer loan approval process application might bundle:

Reviewer: ```suggestion
For instance, a consumer loan approval process application might contain:
```

> - A main BPMN process (e.g., consumer-loan-application.bpmn) to define the workflow.

Reviewer: ```suggestion
- A main BPMN process (e.g., `consumer-loan-application.bpmn`) to define the workflow.
```

- DMN decision tables (e.g., interest-rate-calculation.dmn, credit-score-calculation.dmn) for business rules.

> - Forms (e.g., loan-application-review.form) for user interactions.

Reviewer: ```suggestion
- Forms (e.g., `loan-application-review.form`) for user interactions.
```

> ```

Reviewer: Inside of a typical Java/Maven application the files may be structured as follows:

```suggestion
```
```

consumer-loan-application/
├── src/
│   ├── main/
│   │   ├── java/
│   │   │   └── com/example/loan/
│   │   │   │   └── workers
│   │   │   │   │   ├── UnderwriteLoanWorker.java
│   │   │   │   │   └── ...
│   │   ├── resources/
│   │       ├── consumer-loan-application.bpmn
│   │       ├── dmn/
│   │       │   ├── interest-rate-calculation.dmn
│   │       │   └── credit-score-calculation.dmn
│   │       ├── form/
│   │       │   └── loan-application-review.form
│   │       └── .process-application
│   └── test/
│       └── java/
│           └── ...
├── pom.xml
└── README.md
```

> 

Reviewer: Consider: Should we market some of the features that come with a process application?

```suggestion
### Features

...
```

Author: Yes, will add!

> When opening a file in the modeler the process application will be opened automatically and all other files in the same process application will be shown and can be navigated to.

Reviewer: Independent from the actual folder structure the Modeler will recognize Camunda 8 resources grouped in a process application as soon as the first diagram is opened.

It will also support you when validating between the resources, or when linking resources together.
